vision/use-cases/video-thumbnail-generation/video_thumbnail_generation.ipynb

```diff
@@ -0,0 +1,717 @@
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "35ce6c942d21"
+      },
+      "outputs": [],
+      "source": [
+        "response_text = call_gemini_1_5_pro(\n",
+        "    prompt=basic_prompt,\n",
+        "    gcs_video_path=video_uri,\n",
+        "    model=model,\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "916b62896ee6"
+      },
+      "source": [
+        "### showing JSON output from Gemini"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "2c3e94043a97"
+      },
+      "outputs": [],
+      "source": [
+        "print(json.loads(response_text))"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "a4a72354a934"
+      },
+      "source": [
+        "### displaying thumbnail results with captions"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "b57da95884ee"
+      },
+      "outputs": [],
+      "source": [
+        "display_results(response_text, clip)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "903857971e6b"
+      },
+      "source": [
+        "# Case 2: Using an Advanced Prompt"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "50d6d00f63bd"
+      },
+      "source": [
+        "### Writing an advanced prompt for better thumbnail generation"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "19771c8d3bf3"
+      },
+      "outputs": [],
+      "source": [
+        "advanced_prompt = \"\"\"You are an expert in video content creation and content marketing.\n",
+        "You have the ability to find best thumbnails from a video and provide meaningful and short and catchy captions for them.\n",
+        "Your task is to find the best 4 thumbnails from a given video along with short, and meaningful captions that is good for marketing.\n",
+        "Consider the following rules while generating thubmnails:\n",
+        "\n",
+        "- Thumbnail should have clear focus on the key objects and people, less focus on background\n",
+        "- Thumbnail image should be high quality and bright, avoid blurry images\n",
+        "- Thumbnail image and caption together tell a story\n",
+        "- Thumbnail caption is good for marketing\n",
+        "\"\"\""
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "a22dd4dc041c"
+      },
+      "source": [
+        "### calling Gemini API with advanced prompt"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "32c86bd57475"
+      },
+      "outputs": [],
+      "source": [
+        "response_text2 = call_gemini_1_5_pro(\n",
```

Comment:
Change this variable to be `response_text_advanced` or something similar

Reply:
Done